### ISSUE TITLE
Zero-copy way to make flat tensor into a nested_tensor given a shape

### ISSUE THREAD

**Issue by frankier** [pytorch/pytorch#89124]
### 🚀 The feature, motivation and pitch

If it were possible to create a nested_tensor from a flat tensor with the correct layout, we could always work around elementwise functions which are not (yet) supported by nested_tensor e.g.

```
a_shape = [t.size() for t in a.unbind()]

tensor.nested.nested_tensor_from_values_as(elementwise_function(a.values()), a)

# OR

tensor.nested.nested_tensor_from_values(elementwise_function(a.values()), a_shape)
```

### Alternatives

Copying or trying to work with the flat tensor directly.

### Additional context

I am currently trying to use unsupported loss functions in their unreduced form

**Comment by drisspg** [comment]
Thanks for commenting with this feature request, I am going to add these to this list: https://github.com/pytorch/pytorch/issues/83994
